java/component/keyple-plugin/remotese/src/main/java/org/eclipse/keyple/plugin/remotese/pluginse/VirtualObservableReaderImpl.java

```diff
@@ -110,22 +114,64 @@ public final void clearObservers() {
     }
 
     @Override
-    public void notifySeProcessed() {
-        // TODO Check why this method is empty here
+    public final void notifySeProcessed() {
+        if (forceClosing) {
+            try {
+                // close the physical channel thanks to CLOSE_AFTER flag
+                processSeRequest(null, CLOSE_AFTER);
+                logger.trace(
+                        "Explicit communication closing requested, starting removal sequence.");
+            } catch (KeypleReaderException e) {
+                logger.error("KeypleReaderException while terminating. {}", e.getMessage());
+            }
+        } else {
+            logger.trace("Explicit physical channel closing already requested.");
+        }
     }
 
     @Override
     public void startSeDetection(PollingMode pollingMode) {
-        logger.error(
-                "startSeDetection is not implemented in VirtualObservableReaderImpl, please use the local method");
+        logger.warn(
+                "startSeDetection is not implemented in VirtualObservableReaderImpl, please use the method on the slave node");
     }
 
     @Override
     public void stopSeDetection() {
-        logger.error(
-                "stopSeDetection is not implemented in VirtualObservableReaderImpl, please use the local method");
+        logger.warn(
+                "stopSeDetection is not implemented in VirtualObservableReaderImpl, please use the method on the slave node");
     }
 
+    /*
+     * PACKAGE PRIVATE
+     */
+
+    /**
+     * When an event occurs on the Remote LocalReader, notify Observers
+     *
+     * @param event
+     */
+    void onRemoteReaderEvent(final ReaderEvent event) {
+        final VirtualReaderImpl thisReader = this;
```

Comment:
Line 154 of VirtualObservableReaderImpl.java, should thisReader be defined as a VirtualReaderImpl ? or as a VirtualObservableReaderImpl?
If VirtualObservableReaderImpl, line 158 et 164 the cast would be no more necessary.